build/bin/sage-get-system-packages

```diff
@@ -42,11 +45,25 @@ case "$SYSTEM" in
         fi
         SYSTEM_PACKAGES_FILE_NAMES="distros/$SYSTEM.txt"
         STRIP_COMMENTS="sed s/#.*//;s/\${PYTHON_MINOR}/${PYTHON_MINOR}/g"
+        FROM_PYPROJECT_TOML=0
         COLLECT=echo
         ;;
 esac
-for PKG_BASE in $SPKGS; do
 
+for PKG_BASE in $SPKGS; do
+    if [ $FROM_PYPROJECT_TOML -eq 1 ]; then
+        if [ -f "$SAGE_ROOT/src/pyproject.toml" ]; then
+            # Extract from the "requires" block in src/pyproject.toml
+            REQUIRES=$(sed -n '/requires = \[/,/^\]/p' "$SAGE_ROOT/src/pyproject.toml")
+            # Packages are in the format "'sage-conf ~= 10.3b3',"
+            PACKAGE_INFO=$(echo "$REQUIRES" | sed -n "s/^\s*'\($PKG_BASE.*\)',/\1/p")
```

Comment:
This looks highly suspicious. Is this really how you want us to parse a complex file format like toml?